tests/unit/test_session.py

```diff
@@ -79,3 +84,116 @@ def test_used_scoped_temp_object():
         _PYTHON_SNOWPARK_USE_SCOPED_TEMP_OBJECTS_STRING: False
     }
     assert Session(fake_connection)._use_scoped_temp_objects is False
+
+
+def test_close_exception():
+    fake_connection = mock.create_autospec(ServerConnection)
+    fake_connection._conn = mock.Mock()
+    fake_connection.is_closed = MagicMock(return_value=False)
+    exception_msg = "Mock exception for session.cancel_all"
+    fake_connection.run_query = MagicMock(side_effect=Exception(exception_msg))
+    with pytest.raises(
+        Exception, match=f"Failed to close this session. The error is: {exception_msg}"
+    ):
+        session = Session(fake_connection)
+        session.close()
+
+
+def test_resolve_import_path_ignore_import_path(tmp_path_factory):
+    fake_connection = mock.create_autospec(ServerConnection)
+    fake_connection._conn = mock.Mock()
+    session = Session(fake_connection)
+
+    tmp_path = tmp_path_factory.mktemp("session_test")
+    a_temp_file = tmp_path / "file.txt"
+    a_temp_file.write_text("any text is good")
+    try:
+        # import_path is ignored because file is not a .py file
+        abs_path, _, leading_path = session._resolve_import_path(
+            str(a_temp_file), import_path="a.b"
+        )
+        assert abs_path == str(a_temp_file)
+        assert leading_path is None
+    finally:
+        os.remove(a_temp_file)
+
+
+def test_resolve_package_terms_not_accepted():
+    fake_connection = mock.create_autospec(ServerConnection)
+    fake_connection._conn = mock.Mock()
+    session = Session(fake_connection)
+
+    def get_information_schema_packages(table_name: str):
+        if table_name == "information_schema.packages":
+            result = MagicMock()
+            result.filter().group_by().agg()._internal_collect_with_tag.return_value = (
+                []
+            )
+            return result
+
+    def run_query(sql: str):
+        if sql == "select system$are_anaconda_terms_acknowledged()":
+            return [[False]]
+
+    session.table = MagicMock(name="session.table")
+    session.table.side_effect = get_information_schema_packages
+    session._run_query = MagicMock(name="session._run_query")
+    session._run_query.side_effect = run_query
+    with pytest.raises(
+        ValueError,
+        match="Cannot add package random_package_name because Anaconda terms must be accepted by ORGADMIN to use Anaconda 3rd party packages. Please follow the instructions at https://docs.snowflake.com/en/developer-guide/udf/python/udf-python-packages.html#using-third-party-packages-from-anaconda.",
```

Comment:
this is a very long line. Should we break this down into a smaller one?

Reply:
Broken into multi-lines